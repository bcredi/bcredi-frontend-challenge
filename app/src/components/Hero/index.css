/**
 * Hero
 */

.hero {
  background-position: center;
  background-size: cover;
  width: 40%;
}

.hero__overlay {
  align-items: flex-end;
  background-image: var(--gradient);
  display: flex;
  height: 100%;
  justify-content: flex-end;
  overflow: auto;
  width: 100%;
}

.hero__overlay__content {
  align-items: flex-end;
  color: #fff;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  padding: 60px;
}

.hero__quote {
  font-size: 2rem;
  text-align: right;
}

.hero__name {
  font-weight: bold;
}

.hero__name,
.hero__role {
  font-size: 1.6rem;
  margin: 0 0 var(--gutter);
}

.hero__quote,
.hero__name,
.hero__role {
  color: #fff;
}

@media screen and (max-width: 768px) {
  .hero {
    display: none;
  }
}
