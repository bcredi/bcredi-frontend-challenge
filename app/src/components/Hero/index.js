import React from "react";

// Backgrounds
import BACKGROUND_ONE from "../../assets/images/signup-background-one.jpg";
import BACKGROUND_TWO from "../../assets/images/signup-background-two.jpg";
import BACKGROUND_THREE from "../../assets/images/signup-background-three.jpg";

// Utils
import random from "../../utils/randomBetween";

// Styles
import "./index.css";

// Component
const Hero = ({ children }) => {
  const backgrounds = [BACKGROUND_ONE, BACKGROUND_TWO, BACKGROUND_THREE];
  const index = random(3);
  const background = backgrounds[index];

  return (
    <div className="hero" style={{ backgroundImage: `url(${background})` }}>
      {children}
    </div>
  );
};

export default Hero;
