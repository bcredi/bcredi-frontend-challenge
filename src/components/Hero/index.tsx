import React, { ReactNode } from "react";
import styled from "styled-components";
import heroImage from "../../assets/rawpixel-411166-unsplash@2x.png";

interface props {
  children?: ReactNode;
}

const Hero = ({ children }: props) => {
  return (
    <ImageWrapper className="hero">
      <HeroContent className="hero-content">{children}</HeroContent>
    </ImageWrapper>
  );
};

const ImageWrapper = styled.div`
  height: 100vh;
  width: 43.203125%;
  background-image: linear-gradient(
      rgba(0, 0, 0, 0.2),
      rgba(0, 0, 0, 0) 16%,
      rgba(0, 0, 0, 0) 42%,
      rgba(0, 0, 0, 0.5) 69%,
      rgba(0, 0, 0, 0.8) 96%
    ),
    url(${heroImage});
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
  position: relative;

  @media (max-width: 600px) {
    display: none;
  }
`;

const HeroContent = styled.div`
  color: var(--base-color-white);
  text-align: right;
  width: 66.85%;
  position: absolute;
  right: 7.25%;
  bottom: 4.72%;
`;

export default Hero;
