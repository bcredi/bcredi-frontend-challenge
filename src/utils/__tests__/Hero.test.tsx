import React from "react";
import renderer from "react-test-renderer";
import Hero from "../../components/Hero";
import { shallow } from "enzyme";

describe("<Hero />", () => {
  let component;

  beforeEach(() => {
    component = shallow(<Hero />);
  });

  test("It should mount", () => {
    expect(component.length).toBe(1);
  });

  test("Checks if the component is correctly rendered", () => {
    const component = renderer
      .create(
        <Hero>
          <p>Test</p>
        </Hero>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test("Should render the hero content", () => {
    const wrapper = shallow(
      <Hero>
        <p>Test</p>
      </Hero>
    );
    const text = wrapper.find("p");
    expect(text.text()).toBe("Test");
  });
});
