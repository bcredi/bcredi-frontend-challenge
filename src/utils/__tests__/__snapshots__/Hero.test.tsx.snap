// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> Checks if the component is correctly rendered 1`] = `
<div
  className="sc-bdfBwQ evpleA hero"
>
  <div
    className="sc-gsTCUz mLZhx hero-content"
  >
    <p>
      Test
    </p>
  </div>
</div>
`;

exports[`Checks if the component is correctly rendered 1`] = `
<div
  className="sc-bdfBwQ evpleA hero"
>
  <div
    className="sc-gsTCUz mLZhx hero-content"
  >
    <p>
      Test
    </p>
  </div>
</div>
`;
